lib/cosmos/tools/limits_monitor/limits_monitor.rb

```diff
@@ -815,9 +816,7 @@ def limits_thread
       @limits_thread = Thread.new do
         while true
           break if @cancel_thread
-          Qt.execute_in_main_thread(true) do
-            result, color = @limits_items.process_events()
-          end
+          result, color = @limits_items.process_events()
```

Comment:
Executing this in the main thread causes a brief GUI lockup. I was able to remove it and it all works.